src/mattack_actors.cpp

```diff
@@ -187,7 +187,25 @@ bool mon_spellcasting_actor::call( monster &mon ) const
         return false;
     }
 
-    add_msg( spell_data.message(), mon.disp_name(), spell_data.name() );
+    std::string target_name = "";
+    const monster* target_monster = g->critter_at<monster>( target );
+    if( target_monster ) {
+        target_name = target_monster->disp_name();
+    } else {
+        const npc *target_npc = g->critter_at<npc>( target );
+        if( target_npc ) {
+            target_name = target_npc->disp_name();
+        } else {
+            const avatar *target_avatar = g->critter_at<avatar>( target );
+            if( target_avatar )
+            {
+                target_name = target_avatar->disp_name();
+            }
+        }
+    }
+    if( g->u.sees( target ) ) {
+        add_msg( spell_data.message(), mon.disp_name(), spell_data.name(), _( target_name ) );
```

Comment:
`target_name` has already been translated in `disp_name()`, no need to translate it again.

Reply:
the `creature::disp_name` returns a `std::string` and not a `translation` though? that is confusing :(

Reply:
`translation` only wraps the translatable message (and an optional context) for later translation. Most of the game's code is still using std::string to pass both translated and untranslated strings, and in this case the result from `disp_name` is already translated.